plugins/modules/omevv_firmware.py

```diff
@@ -0,0 +1,1224 @@
+        :type module: Any
+        :return: None
+        """
+
+    def get_payload_details(self, host_id):
+        """
+        Retrieves the payload details for the firmware update.
+
+        Args:
+            host_id (str): The ID of the host.
+
+        Returns:
+            dict: The payload details for the firmware update.
+        """
+        device_id = host_id
+        parameters = self.module.params
+        target_list = parameters['targets']
+        payload = {"firmware": {"targets": []}}
+        firmware = payload["firmware"]
+
+        payload["firmware"].update(self.set_firmware(firmware, parameters))
+        payload = self.set_schedule(payload, parameters)
+        payload = self.set_job_details(payload, parameters)
+        payload["firmware"].update(self.add_targets(firmware, target_list, device_id))
+
+        return payload
+
+    def set_firmware(self, firmware, parameters):
+        """
+        Adds optional fields to the firmware dictionary based on the
+        parameters.
+
+        Args:
+            firmware (dict): The firmware dictionary.
+            parameters (dict): The parameters dictionary.
+
+        Returns:
+            None
+        """
+        optional_fields = [
+            ('check_vSAN_health', 'checkvSANHealth'),
+            ('delete_job_queue', 'deleteJobsQueue'),
+            ('drs_check', 'drsCheck'),
+            ('enter_maintenance_mode_options', 'enterMaintenanceModeOption'),
+            ('enter_maintenance_mode_timeout', 'enterMaintenanceModetimeout'),
+            ('evacuate_VMs', 'evacuateVMs'),
+            ('exit_maintenance_mode', 'exitMaintenanceMode'),
+            ('maintenance_mode_count_check', 'maintenanceModeCountCheck'),
+            ('reboot_options', 'rebootOptions'),
+            ('reset_idrac', 'resetIDrac'),
+        ]
+
+        for param_key, firmware_key in optional_fields:
+            # if parameters.get(param_key) is not None:
+            firmware[firmware_key] = parameters.get(param_key)
+        return firmware
+
+    def set_schedule(self, payload, parameters):
+        """
+        Sets the schedule for the payload.
+
+        Args:
+            payload (dict): The payload dictionary.
+            parameters (dict): The parameters dictionary.
+
+        Returns:
+            None
+        """
+        payload["schedule"] = {"runNow": parameters.get('run_now')}
+
+        if not payload["schedule"]["runNow"]:
+            payload["schedule"]["dateTime"] = parameters.get('date_time')
+        return payload
+
+    def set_job_details(self, payload, parameters):
+        """
+        Sets the job details in the payload dictionary.
+
+        Args:
+            payload (dict): The payload dictionary.
+            parameters (dict): The parameters dictionary.
+
+        Returns:
+            None
+        """
+        if parameters.get('job_description'):
+            payload["jobDescription"] = parameters.get('job_description')
+        else:
```

Comment:
Else part is not required , parameters.get('job_description') will return None only if job_description is not passed by user.

Reply:
This is required